### PR TITLE
Add apply_meta_as_tags macro

### PR DESCRIPTION
**Pull request by jamesweakley**
This PR adds the apply_meta_as_tags as discussed here: https://github.com/dbt-labs/dbt-snowflake/issues/104

Documentation is available in the readme, I opted for a yaml-driven approach.

Tested on a sample model with dbt 1.3.0.

### CODE REVIEW

**Comment by MartinGuindon** [comment]
Thank you for your contribution @jamesweakley. I will aim to merge this in the next few days!

**Comment by gmatsonANSYS** [comment]
@jamesweakley I tested your macro using the yaml driven approach and the macro only executes if a table and column tag are set on a model. Are you able to have tags created/set using only the column meta tags?

the log file shows  " meta': {'database_tags': None}, 'transient': False}" and skips

**Comment by jamesweakley** [comment]
@gmatsonANSYS good pickup, I hadn't tried that scenario. Will take a look today

**Comment by jamesweakley** [comment]
@gmatsonANSYS it should now work even if you've only defined column tags

**Comment by gmatsonANSYS** [comment]
Thanks James, the updates worked as expected. The only tweaks I had to make were around the existing tag filters:

Converted the table and column name to upper - similar to the tag_name pipe - to return matches.
`{%- set existing_tag_for_table = existing_tags|selectattr('0','equalto','TABLE')|selectattr('1','equalto',table_name|upper)|selectattr('2','equalto',column_name|upper)|selectattr('3','equalto',tag_name|upper)|list -%}`

Updated to the fifth column index for the tag value comparison
`{% if existing_tag_for_table|length > 0 and existing_tag_for_table[0][4]==desired_tag_value %}`

**Comment by jamesweakley** [comment]
@gmatsonANSYS thanks for the code snippets, I've pushed some more changes and tested various matching/non-matching scenarios and it seems consistent now

**Comment by jamesweakley** [comment]
@gmatsonANSYS hopefully this all works in your test scenario now?

**Comment by gmatsonANSYS** [comment]
@jamesweakley  runs smooth as silk. Thank you!

**Comment by MartinGuindon** [comment]
Awesome, if there are no further updates to this PR I'll be merging it shortly.

File: README.md [hunk 1]
```diff
@@ -104,6 +104,48 @@ Call the macro as an [operation](https://docs.getdbt.com/docs/using-operations):
...
+
+#### Arguments
+* `results` (required): The [on-run-end context object](https://docs.getdbt.com/reference/dbt-jinja-functions/on-run-end-context).
+
+#### Usage
+
+First, configure your dbt model to have the 'database_tags' meta property as shown (tag examples borrowed from [here](https://docs.snowflake.com/en/user-guide/tag-based-masking-policies.html)):
+
+```
+schema.yma
```
Comment (MartinGuindon) [inline README.md]: I assume this should be `# schema.yml`?

**Review by MartinGuindon — CHANGES_REQUESTED** [review]
@jamesweakley The macro should be documented in macros.yml, and there's a typo in the README. Other than that, looks pretty good to me . Once fixed I'll merge and issue a new version.

**Comment by jamesweakley** [comment]
@MartinGuindon Thanks, I've pushed those changes now

**Review by MartinGuindon — APPROVED** [review]

**Pull request merged**